### DISCUSSION TITLE
Correct defination  of constructor in solidity

### DISCUSSION THREAD

**Discussion by LavleshJ** [Cyfrin/foundry-full-course-cu#480] [Q&A] [answered]
###  *Correct definition of constructor in solidity

I know we define constructor as follows:
```
contract Person {
    
    address public owner;

    // Constructor
    constructor() { // with constructor keyword
        owner = msg.sender;
    }

    // Other functions and variables...
}

```

Can we define the constructor with contract name using function keyword as shown below ?
```
contract Person {
   
    address public owner;

    // Constructor
    function Person() public payable {
       owner = msg.sender;
    }

    // Other functions and variables...
}

```

**Comment by Naman1729** [comment]
Hey @LavleshJ, based on my knowledge of Solidity, it is not possible to define the constructor with the contract name using the function keyword in Solidity. **Constructor** is a special function that is automatically executed only once during the contract deployment .

**Comment by 0xusmanf** [comment, marked as answer]
Prior to version 0.4.22 of Solidity, constructors were defined as functions with the same name as the contract, but this syntax is deprecated and not allowed since version 0.5.0.

**Reply by LavleshJ** [reply]
Oh, okay, though it is deprecated but, will it work with solidity version `Prior to version 0.4.22` ?

**Reply by 0xusmanf** [reply]
Yes, it will work.
